arch/z80/src/common/z80_doirq.c

```diff
@@ -47,6 +47,7 @@
 FAR chipreg_t *z80_doirq(uint8_t irq, FAR chipreg_t *regs)
 {
   struct tcb_s **running_task = &g_running_tasks[this_cpu()];
+  struct tcb_s *tcb;
 
   if (*running_task != NULL)
```

Comment:
do we still need check NULL. should we use running_task at line 106

Reply:
There is no need to check if *running_task != NULL here, as z80_doirq only handles interrupts and not system calls. We can directly use *running_task later on. We plan to address this with an additional patch

Reply:
ok, please create a new patch to fix ALL places.